src/main/java/io/lettuce/core/output/JsonValueListOutput.java

```diff
@@ -0,0 +1,49 @@
+package io.lettuce.core.output;
+
+import io.lettuce.core.codec.RedisCodec;
+import io.lettuce.core.json.JsonValue;
+import io.lettuce.core.json.JsonParser;
+import io.lettuce.core.json.JsonParserRegistry;
+
+import java.nio.ByteBuffer;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * {@link List} of string output.
+ *
+ * @param <K> Key type.
+ * @param <V> Value type.
+ * @author Will Glozer
+ * @author Mark Paluch
+ */
+public class JsonValueListOutput<K, V> extends CommandOutput<K, V, List<JsonValue<K, V>>> {
+
+    private boolean initialized;
+
+    private final JsonParser<K, V> parser;
+
+    public JsonValueListOutput(RedisCodec<K, V> codec) {
+        super(codec, Collections.emptyList());
+
+        parser = JsonParserRegistry.getJsonParser(codec);
+    }
+
+    @Override
+    public void set(ByteBuffer bytes) {
+        if (!initialized) {
+            multi(1);
+        }
+
+        output.add(parser.createJsonValue(bytes));
```

Comment:
Aren't we supposed to use the codec to decode the raw command output to "something" **decoded** here?

It seems that the execution chain is wrong; it should be:
1. codec is used to decode the raw response 
2. Json parser tries to parse what codec decoded

This way, the application can have custom codecs that do decompression or decryption with a custom codec.

Reply:
I think I got you confused with the naming. Two things to consider:
- the "parser" is used for both parsing a String (containing JSON) and creating an object model (JsonValue, JsonObject, JsonArray) out of it, that the app can use to modify this JSON; but also to provide the implementation of these interfaces, based on the currently set up parser (as we can have custom parsers). The DefaultParser works with Jackson, but the user might want to replace it with another parser
- the `JsonValueListOutput` is called inside the event loop and any type of processing (including simply decoding the JSON document) could incur performance degradation on the event loop. An example situation is a large JSOn file and a Codec that does encryption/compression.

The current model avoids this by wrapping the result inside an UnproccesedJsonValue and returning that to the user thread. Then, depending on the pattern the user wants to use the app, they could either call `getValue()` which would apply the codec associated with the connection and return a value of the specific type or call any of the methods `.asJsonObject()` / `is.JsonObject()` / `.isJsonArray()`/ etc. that would lazily parse the result (again using the codec) and provide an object model to work with the JSON document.

In any case no processing is done to the raw data in the event loop.